### DISCUSSION TITLE
Questions About Session Transfer and Compatibility Between Servers and Engines

### DISCUSSION THREAD

**Discussion by TheOneironaut** [devlikeapro/waha#657] [Q&A] [answered]
1. I have sessions on one server. Can I transfer them to another server? What will happen if I run the same session simultaneously on two different servers?

2. Is it possible to transfer a session from noweb to webjs and vice versa?

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment, marked as answer]
Hi! 

> Can I transfer them to another server?

Yes, you need to stop docker container and copy `.sessions` or `sessions` folder to the new server if you're using default installation

> What will happen if I run the same session simultaneously on two different servers?

WhatsApp doesn't allow it - it'll continue restarting sessions on both servers due to "Conflict" reason on websocket. Right now WAHA doesn't have any detection for that - this is why we're suggesting using horizontal scaling using sharding (aka servers have different databases)
https://waha.devlike.pro/blog/waha-scaling-how-to-handle-500-sessions/#horizontal-scaling---sharding

> Is it possible to transfer a session from noweb to webjs and vice versa?

Nope, and we don't have plans to implement that yeat.

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
